File: spring-java/src/main/java/com/bsthun/w/chxpos/apidemo/utils/MySqlConnector.java
package com.bsthun.w.chxpos.apidemo.utils;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class MySqlConnector {
	private static final Logger logger = LoggerFactory.getLogger(MySqlConnector.class);
	private static Connection connection;
	
	public MySqlConnector() throws SQLException {
		reconnect();
	}
	
	public static Connection getConnection() throws SQLException {
		if (connection.isClosed())
			reconnect();
		return connection;
	}
	
	private static void reconnect() throws SQLException {
		connection = DriverManager.getConnection(Secret.MYSQL_URL, Secret.MYSQL_USERNAME, Secret.MYSQL_PASSWORD);
		logger.info("Reconnected to MySQL Server");
	}
}

File: spring/src/main/kotlin/com/bsthun/w/chxpos/apidemo/utils/MySqlConnector.kt
package com.bsthun.w.chxpos.apidemo.utils

import org.apache.commons.dbcp.BasicDataSource
import org.slf4j.LoggerFactory
import java.sql.Connection

object MySqlConnector {
	private val dataSource = BasicDataSource()
	
	init {
		dataSource.url = SecretProperties.MYSQL_URL
		dataSource.username = SecretProperties.MYSQL_USERNAME
		dataSource.password = SecretProperties.MYSQL_PASSWORD
	}
	
	val connection: Connection
		get() = dataSource.connection
}